lib/Backends/CPU/libjit/libjit.cpp

```diff
@@ -425,16 +425,58 @@ static void libjit_gatherranges(T *output, U *lengths, const T *data,
 }
 
 template <typename T>
-static void libjit_scatterassign(T *data, const size_t *indices,
-                                 const T *slices, size_t numIndices,
-                                 size_t sliceSize) {
+static void libjit_scatterdatacopy(T *data, const size_t *dataDims,
+                                   const size_t *indices, const T *slices,
+                                   size_t numIndices, size_t indexSize,
+                                   size_t sliceSize) {
   for (size_t i = 0; i < numIndices; i++) {
-    size_t destDataIdx = indices[i];
+    size_t destDataIdx = indices[i * indexSize];
```

Comment:
This seems like it's still generalized past 2D right? Maybe simplify it given we only support 2D?

Reply:
This code was initially was done for 2D, and it also worked for 1D as `indexSize` as computed as `indices->size() / numIndices`, and for 1D case it was 1. Now I made change in LLVM IR gen to clearly take these values from indices->dims()[0], indices->dims()[1] to make this logic more transparent. But I don't see how we can simplify code here. What's your idea?

Reply:
Nevermind I think it's good.